features/step_definitions/websocket-steps.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+const should = require('should');
+const { Given, Then, When } = require('cucumber');
+const ws = require('ws');
+
+Given('I open a new local websocket connection', async function () {
+  this.props.client = new ws('ws://localhost:7512');
+  const delay = ms => new Promise(resolve => setTimeout(resolve, ms));
```

Comment:
Why are you waiting 1 sec here? 1 second is huge, it will the slow down all the tests

Reply:
I removed it, that was just a test